src/tools/rados/rados.cc

```diff
@@ -2896,8 +2959,11 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
       goto out;
     }
   } else if (strcmp(nargs[0], "rm") == 0) {
-    if (!pool_name || nargs.size() < 2)
-      usage_exit();
+    if (!pool_name || nargs.size() < 2) {
+      usage(cerr);
+      ret = -1;
+      goto out;
```

Comment:
the whole point of `goto out` is to `delete formatter`. but guess we can fix this by managing it using a unique_ptr<> in a follow-up PR.

Reply:
@tchaikov updated with unique_ptr used